src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -49,9 +49,11 @@
 
 /* -- Default credentials. -- */
 
-static int g_compute_engine_detection_done = 0;
-static int g_need_compute_engine_creds = 0;
+static int g_metadata_server_detection_done = 0;
```

Comment:
Could you please add a comment explaining why this variable is sticky for positive results, but not negative? It's totally the right thing to do, but I'm sure posterity would love to know.

Reply:
Good suggestion. A comment is added. PTAL.